src/contracts/Protocol/QuestionStateController.sol

```diff
@@ -92,8 +94,14 @@ contract QuestionStateController is IQuestionStateController, Ownable, OnlyApi {
         return votes[questionId].totalVoteCount;
     }
 
-    function getQuestionsByState(STATE currentState) public view returns (QuestionData[] memory) {
-        return questionByState[currentState];
+    function getQuestionsByState(STATE currentState, uint256 currentQuestionId) public view returns (QuestionData[] memory) {
```

Comment:
we create a new array in memory because we know the amount we need to loop through by the second argument which is the latest or current question ID 
we loop backwards but have to create array from 0 so that's why we have i and j